pkg/plugins/diff/delta_diff_server/src/utils.rs

```diff
@@ -31,48 +30,39 @@ const TIMESTAMP_KEY: &str = "timestamp";
 
 pub(crate) fn construct_storage_config(config: GatewayConfig) -> HashMap<String, String> {
     let mut s3_config: HashMap<String, String> = HashMap::new();
-    s3_config.insert("AWS_ACCESS_KEY_ID".to_string(), config.key);
-    s3_config.insert("AWS_ENDPOINT_URL".to_string(), config.endpoint);
-    s3_config.insert("AWS_S3_ADDRESSING_STYLE".to_string(), "path".to_string());
-    s3_config.insert("AWS_SECRET_ACCESS_KEY".to_string(), config.secret);
-    s3_config.insert("AWS_STORAGE_ALLOW_HTTP".to_string(), "true".to_string());
+    s3_config.insert(s3_storage_options::AWS_ACCESS_KEY_ID.to_string(), config.key);
```

Comment:
I'd expect these options to be usable directly somehow, it's really strange to have a non-string type that cannot be used.  Can we make `s3_config` accept a `s3_storage_option` or whatever that type is?